File: SQL/Expressions/SQLAggregateExpression.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2012. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLAggregateExpression
        Inherits SQLFunctionExpression

        Public Sub New(ByVal aggregate As AggregateFunction, ByVal fieldName As String)

            Me.New(aggregate, New SQLFieldExpression(fieldName))

        End Sub

        Public Sub New(ByVal aggregate As AggregateFunction, ByVal expression As SQLExpression)

            MyBase.New(Misc.SQLConvertAggregate(aggregate), expression)

        End Sub

    End Class

End Namespace

File: SQL/Expressions/SQLAggregateExpression.cs
// _________________________________________________________________________
//
//  © Hi-Integrity Systems 2012. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// 
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//	    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// _________________________________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLAggregateExpression : SQLFunctionExpression
	{
		private AggregateFunction aggregate;

		public SQLAggregateExpression(AggregateFunction aggregate, string fieldName) 
            : this(aggregate, new SQLFieldExpression(fieldName))
		{
		}
			
		public SQLAggregateExpression(AggregateFunction aggregate, SQLExpression expression) 
            : base(expression)
		{
			this.aggregate = aggregate;
		}

		public AggregateFunction Aggregate
		{
			get
			{
				return aggregate;
			}
		}

		internal override string SQL(Serializers.Serializer serializer)
		{
			return serializer.SerializeAggregateExpression(this);
		}
	}
}
